app/src/main/java/org/mozilla/fenix/settings/TrackingProtectionFragment.kt

```diff
@@ -143,7 +143,7 @@ class TrackingProtectionFragment : PreferenceFragmentCompat() {
         context?.components?.let {
             val policy = it.core.createTrackingProtectionPolicy()
             it.useCases.settingsUseCases.updateTrackingProtection.invoke(policy)
-            it.useCases.sessionUseCases.reload.invoke()
+            context?.settings()?.shouldReload = true
```

Comment:
I wonder why this is a setting (do we really need to persist it?) and why we could not hide this flag inside `GeckoEngineSession` - can we not handle this internally (maybe hiding some of it behind a usecase if needed)?

Reply:
> I wonder why this is a setting (do we really need to persist it?)

Good catch, I think we could do it without persisting the setting, as if the app get closed the next time the user opens a session it will reload.

> Why we could not hide this flag inside GeckoEngineSession - can we not handle this internally (maybe hiding some of it behind a usecase if needed)?

I brought up the patch to momentarily stop the freezing until a more permanent solution is created from GV. They submitted a patch but I'm not quite sure if it will permanently fix the issue https://bugzilla.mozilla.org/show_bug.cgi?id=1609701#c11